src/SharingRole.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Owncloud\OcisPhpSdk;
+
+use OpenAPI\Client\Model\UnifiedRoleDefinition;
+use Owncloud\OcisPhpSdk\Exception\InvalidResponseException;
+
+/**
+ * class to define a role of a user or group in a share
+ * every role contains specific permissions
+ */
+class SharingRole
+{
+    private string $id;
+    private string $displayName;
+    private string $description;
+    private int $weight;
+
+    /**
+     */
+    public function __construct(UnifiedRoleDefinition $apiRole)
+    {
+        $this->id = !is_string($apiRole->getId()) ?
+            throw new InvalidResponseException(
+                "Invalid id returned for sharing role '" . print_r($apiRole->getId(), true) . "'"
+            )
+            : $apiRole->getId();
+        $this->displayName = !is_string($apiRole->getDisplayName()) ?
+            throw new InvalidResponseException(
+                "Invalid display name returned for sharing role '" .
+                print_r($apiRole->getId(), true) .
+                "'"
+            )
+            : $apiRole->getDisplayName();
+        $this->description = !is_string($apiRole->getDescription()) ? "" : $apiRole->getDescription();
+        $this->weight = (int)$apiRole->getAtLibreGraphWeight();
+    }
+
+    /**
+     * @return string
+     */
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDisplayName(): string
+    {
+        return $this->displayName;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDescription(): string
+    {
+        return $this->description;
+    }
+
+    /**
+     * @return int
+     */
+    public function getWeight(): int
+    {
+        return $this->weight;
+    }
```

Comment:
just curious what is use of weight?

Reply:
this is to help a client to sort the roles, e.g. `viewer` might have to be shown before `editor`, so sorting alphabetically does not make sense, so a client can sort by `weight`